client/lib/query-manager/with-query-manager.js

```diff
@@ -0,0 +1,35 @@
+/**
+ * Apply a `callback` transformation to a `QueryManager` instance in `state[ siteId ]` and
+ * return a state object updated with its result.
+ * If the `state[ siteId ]` object doesn't exist, it's created with the `create` function.
+ * If `create` is `false`, the `callback` will not be applied (there is no `QueryManager` to
+ * apply it to after all) and unchanged state will be returned.
+ *
+ * @param {object} state State object
+ * @param {string} siteId State key
+ * @param {Function | boolean} create Optional create callback
```

Comment:
Nit: I think I would prefer `Function | null` as the accepted API here. `boolean` seems too wide, what would `true` mean, for example?

Reply:
In TypeScript the type of this parameter would be `Function | false`, but JSDoc ESLint rules don't accept `false` as a type name 🙁

I can change the "no create" option from `false` to `null`, just wanted to point out we're doing it just to please the imperfect JSDoc linter.

Reply:
Another idea: swap the order of the "update" and "create" callbacks. The "update" callback is always there, the "create" one is optional and we can omit it.
```js
withQueryManager(
  state,
  siteId,
  ( manager ) => manager.receive( action.posts ),
  () => new PostQueryManager()
);
```
or
```js
withQueryManager(
  state,
  siteId,
  ( manager ) => manager.remove( action.postId )
);
```
What do you think?

Reply:
> just wanted to point out we're doing it just to please the imperfect JSDoc linter

We _frequently_ disable the jsdoc linter in Gutenberg, FWIW, so if you wanted to use `Function | false` then go for it, I'm all for ignoring that linting rule when it comes to valid TypeScript types.

Reply:
I like the version to make it optional too, I say go for that if you're up for it.

Reply:
Did the parameter swapping and going to merge now.